akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/cookbook/RecipeReduceByKeyTest.java

```diff
@@ -46,27 +53,24 @@ public void work() throws Exception {
 
         //#word-count
         final int MAXIMUM_DISTINCT_WORDS = 1000;
-
-        // split the words into separate streams first
-        final Source<Pair<String, Source<String, BoxedUnit>>, BoxedUnit> wordStreams = words
-          .groupBy(i -> i);
-
-        // add counting logic to the streams
-        Source<Future<Pair<String, Integer>>, BoxedUnit> countedWords = wordStreams.map(pair -> {
-          final String word = pair.first();
-          final Source<String, BoxedUnit> wordStream = pair.second();
-          return wordStream.runFold(
-            new Pair<>(word, 0),
-            (acc, w) -> new Pair<>(word, acc.second() + 1), mat);
-        });
-
-        // get a stream of word counts
-        final Source<Pair<String, Integer>, BoxedUnit> counts = countedWords
-          .buffer(MAXIMUM_DISTINCT_WORDS, OverflowStrategy.fail())
-          .mapAsync(4, i -> i);
+        
+        final Source<Pair<String, Integer>, BoxedUnit> counts = words
+            // split the words into separate streams first
+          .groupBy(MAXIMUM_DISTINCT_WORDS, i -> i)
+          // add counting logic to the streams
+          .fold(new Pair<>("", 0), (pair, elem) -> new Pair<>(elem, pair.second() + 1))
```

Comment:
I am a big fan of `of`. I think we should add it (once streams are on master, that'll be much simpler)